pkg/router/template/router.go

```diff
@@ -865,17 +870,30 @@ func generateDestCertKey(config *ServiceAliasConfig) string {
 
 type endpointsCounter func(key string) int32
 
-// getServiceUnits returns a map of service keys to their weights.
-// The requests are loadbalanced among the services referenced by the route.
+// getServiceUnits returns a map of service keys to their weights
+// and a map of service keys to number of endpoints in the service.
+//
+// The requests are load balanced among the services referenced by the route.
 // The weight (0-256, default 1) sets the relative proportions each
-// service gets (weight/sum_of_weights fraction of the requests).
+// service gets (weight/sum_of_weights) fraction of the requests.
 // When the weight is 0 no traffic goes to the service. If they are
 // all 0 the request is returned with 503 response.
-// For each service, the requests are distributed among the endpoints.
-// Each endpoint gets weight/numberOfEndpoints portion of the requests.
+// For each service, the requests are distributed among the service's
+// endpoints.
+//
+// The service weight is distributed among the endpoints. Since the
+// minimum weight is 1 it is possible that there are too many endpoints
+// for the service weight. E.g, service weight 5 and 9 endpoints.
+// In this case 5 endpoints will get 1 and 4 will get 0. Otherwise,
```

Comment:
We have two choices, and we can emphasize on only one if the weight assigned is too less compared to number of available endpoints. 
1. Keep equitable distribution on pods of a service (i.e. do not mark any pods to weight 0, by design)
2. Respect the A/B ratio as directed by the user

Currently, we do option 2. This algorithm changes our behaviour to 1.
Maybe we can leave this as choice. But this seemingly advanced choice will need to be addressed.